pkg/dronereceiver/factory.go

```diff
@@ -2,53 +2,68 @@ package dronereceiver
 
 import (
 	"context"
-	"time"
 
 	"github.com/grafana/grafana-ci-otel-collector/dronereceiver/internal/metadata"
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/receiver"
 	"go.opentelemetry.io/collector/receiver/scraperhelper"
 )
 
+const (
+	defaultBindEndpoint = "0.0.0.0:3333"
+	defaultPath         = "/drone/webhook"
+)
+
 func createDefaultConfig() component.Config {
 	cfg := scraperhelper.NewDefaultScraperControllerSettings(metadata.Type)
-	cfg.CollectionInterval = 10 * time.Second
 
 	return &Config{
 		ScraperControllerSettings: cfg,
 		MetricsBuilderConfig:      metadata.DefaultMetricsBuilderConfig(),
-		WebhookConfig: WebhookConfig{
-			Endpoint: "/drone/webhook",
-			Port:     3333,
+		ServerConfig: confighttp.ServerConfig{
+			Endpoint: defaultBindEndpoint,
 		},
+		Path:   defaultPath,
+		Secret: "",
 	}
 }
 
 func NewFactory() receiver.Factory {
 	return receiver.NewFactory(
 		metadata.Type,
 		createDefaultConfig,
-		receiver.WithTraces(createTraceReceiver, metadata.TracesStability),
-		receiver.WithMetrics(createMetricsReceiver, metadata.MetricsStability),
-		receiver.WithLogs(createLogsReceiver, metadata.LogsStability))
+		receiver.WithTraces(newTracesReceiver, metadata.TracesStability),
+		receiver.WithMetrics(newMetricsReceiver, metadata.MetricsStability),
+		receiver.WithLogs(newLogsReceiver, metadata.LogsStability))
 }
 
-func createTraceReceiver(_ context.Context, set receiver.CreateSettings, cfg component.Config, consumer consumer.Traces) (receiver.Traces, error) {
-	r, err := getOrAddReceiver(set, cfg)
+func newTracesReceiver(_ context.Context, set receiver.CreateSettings, cfg component.Config, consumer consumer.Traces) (receiver.Traces, error) {
+	rCfg := cfg.(*Config)
+	var err error
+
+	if consumer == nil {
+		return nil, component.ErrNilNextConsumer
+	}
+
+	r := receivers.GetOrAdd(cfg, func() component.Component {
+		var rcv component.Component
+		rcv, err = newReceiver(set, rCfg)
+		return rcv
+	})
```

Comment:
wondering whether we should check for `r` value being nil here

Reply:
if `err` is not nil `r` cannot possibly be nil (according to `newReceiver` implementation). maybe we can add more coverage for `newReceiver` to check that is actually true (now we are only testing the happy path) I personally wouldn't introduce more code that cannot really be reached, i'd rather write more test to validate and lock the behavior. wdyt?

Reply:
yeah that sounds good!